Repository: plone.app.users


Branch: refs/heads/master
Date: 2016-04-30T23:01:26+02:00
Author: Maurits van Rees (mauritsvanrees) <maurits@vanrees.org>
Commit: https://github.com/plone/plone.app.users/commit/caa6a2f9c33ed3fa01d8e24955aa18701844207a

Fixed KeyError email on personal preferences form. (#58)

This could happen when email is used as login name.

Fixes https://github.com/plone/plone.app.users/issues/56
and https://github.com/plone/Products.CMFPlone/issues/1146

Files changed:
M CHANGES.rst
M plone/app/users/browser/account.py

diff --git a/CHANGES.rst b/CHANGES.rst
index 5654679..5a81de2 100644
--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -6,6 +6,12 @@ CHANGES
 
 Fixes:
 
+- Fixed KeyError email on personal preferences form.  This could
+  happen when email is used as login name.  Fixes
+  https://github.com/plone/plone.app.users/issues/56 and
+  https://github.com/plone/Products.CMFPlone/issues/1146
+  [maurits]
+
 - Ensured partial searching utility for users in 'Search for users' page
   Fixes https://github.com/plone/Products.CMFPlone/issues/1499
   [kkhan]
diff --git a/plone/app/users/browser/account.py b/plone/app/users/browser/account.py
index 411e335..bd8027a 100644
--- a/plone/app/users/browser/account.py
+++ b/plone/app/users/browser/account.py
@@ -206,8 +206,10 @@ def handleSave(self, action):
         CheckAuthenticator(self.request)
         data, errors = self.extractData()
 
-        # extra validation for email
-        self.validate_email(action, data)
+        # Extra validation for email, when it is there.  email is not in the
+        # data when you are at the personal-preferences page.
+        if 'email' in data:
+            self.validate_email(action, data)
 
         if action.form.widgets.errors:
             self.status = self.formErrorsMessage


